Sources/CoreMetrics/Metrics.swift

```diff
@@ -488,6 +580,74 @@ public protocol MetricsFactory {
     func destroyTimer(_ handler: TimerHandler)
 }
 
+/// Wraps a CounterHandler, adding support for incrementing by floating point values by storing an accumulated floating point value and recoding increments to the underlying CounterHandler after crossing integer boundaries.
+private class AccumulatingRoundingFloatingPointCounter: FloatingPointCounterHandler {
+    private static let kAutoStateResetThreshold = Double(Int64.max - (2 << 8))
+
+    private var lock = Lock()
+    private var rawValue: Double = 0
+    private var roundedValue: Int64 = 0
+    private(set) var counterHandler: CounterHandler
+
+    init(counterHandler: CounterHandler) {
+        self.counterHandler = counterHandler
+    }
+
+    func increment(by amount: Double) {
+        assert(amount < AccumulatingRoundingFloatingPointCounter.kAutoStateResetThreshold)
+        self.lock.withLockVoid {
+            let incremented = rawValue + amount
+            if incremented < AccumulatingRoundingFloatingPointCounter.kAutoStateResetThreshold {
+                self.rawValue = incremented
+            } else {
+                self.rawValue = amount
+                self.roundedValue = 0
+            }
```

Comment:
in other places (e.g. Timer) we compare to Int.max and continue to emit max instead of rolling to 0. not sure what a better approach in this case.

Reply:
also we should add unit test to test normal + overflow behaviors

Reply:
this defers that behavior to the underlying `CounterHandler` and just ensures that the computation in `increment` won't trap or overflow. though looking at this closer, I think the logic is wrong, hmm...